scripts/pod-mon.sh

```diff
@@ -22,18 +22,57 @@ log () {
   fi
 }
 
-# get cpu/mem usage
-top=$(top -b -n 1 -H -o %CPU | head -n 15)
-free=$(free -h)
-# get disk usage
-df=$(df -h)
-tmpusage=$(du -ch --max-depth=1 /tmp | sort -hr)
-
-log "System usage --------"
-log "$top"
-log "Mem usage --------"
-log "$free"
-log "Disk usage --------"
-log "$df"
-log "/tmp usage --------"
-log "$tmpusage"
+# List of process names to monitor
+processes=("MistController" "MistProcLivepeer" "MistUtilLoad" "MistOutWebRTC" "MistInDTSC" "MistOutDTSC" "MistOutFLV" "MistInFLV" "MistInBuffer" "MistOutHTTPTS" "catalyst-api" "catalyst-uploader")
+
+# Function to calculate CPU and MEM usage for a specific process
+calculate_usage() {
+  local process_name=$1
+
+  # Get the output of ps aux filtered by the process name, skipping the header row and capturing only the CPU and MEM columns
+  ps aux | grep "$process_name" | grep -v grep | awk '{
+    cpu+=$3; mem+=$4; count+=1; if ($3 > max_cpu) max_cpu=$3
+  } END {
+    print cpu, mem, count, max_cpu
+  }'
+}
+
+# Get the number of CPU cores
+num_cores=$(grep -c ^processor /proc/cpuinfo)
+
+# Initialize the output as an empty string
+output="["
+
+# Loop through each process and calculate the usage
+for process in "${processes[@]}"; do
+  output_data=$(calculate_usage "$process")
+
+  cpu=$(echo "$output_data" | awk '{print $1}')
+  mem=$(echo "$output_data" | awk '{print $2}')
+  count=$(echo "$output_data" | awk '{print $3}')
+  max_cpu=$(echo "$output_data" | awk '{print $4}')
+
+  # Set defaults if count is not a number
+  if ! [[ "$count" =~ ^[0-9]+$ ]]; then
+    count=0
+  fi
+
+  if [ "$count" -gt 0 ]; then
+    avg_cpu_per_core=$(awk "BEGIN {print $cpu / $num_cores}")
+    avg_mem=$(awk "BEGIN {print $mem / $count}")
+  else
+    avg_cpu_per_core=0
+    avg_mem=0
+    max_cpu=0
+  fi
```

Comment:
Somehow [avg is higher that max](https://eu-metrics-monitoring.livepeer.live/grafana/explore?schemaVersion=1&panes=%7B%227ac%22%3A%7B%22datasource%22%3A%22P8E80F9AEF21F6940%22%2C%22queries%22%3A%5B%7B%22refId%22%3A%22A%22%2C%22expr%22%3A%22%7Bapp%3D%5C%22catalyst%5C%22%2C+region%3D%5C%22sto%5C%22%2C+node_name%3D%5C%22dp3289%5C%22%7D+%7C%7E+%60avg_cpu_per_core%5C%22%3A+2.%5C%5C.%60%22%2C%22queryType%22%3A%22range%22%2C%22datasource%22%3A%7B%22type%22%3A%22loki%22%2C%22uid%22%3A%22P8E80F9AEF21F6940%22%7D%2C%22editorMode%22%3A%22code%22%7D%5D%2C%22range%22%3A%7B%22from%22%3A%221723036897173%22%2C%22to%22%3A%221723067492115%22%7D%7D%7D&orgId=1).
```
{"process": "MistInDTSC", "avg_cpu_per_core": 27.0275, "max_cpu_per_core": 16.0, "avg_mem_per_process": 0},
```
I think it should be written in python (psutil) or golang.